editoast/src/main.rs

```diff
@@ -43,14 +44,23 @@ async fn run() -> Result<(), Box<dyn Error>> {
     }
 }
 
-async fn runserver(args: RunserverArgs, pg_config: PostgresConfig) -> Result<(), Box<dyn Error>> {
+async fn runserver(
+    args: RunserverArgs,
+    pg_config: PostgresConfig,
+) -> Result<(), Box<dyn Error + Send + Sync>> {
     let conn = PgConnection::establish(&pg_config.url()).expect("Error while connecting DB");
     let infras = Infra::list(&conn);
+    let infras_clone = infras.clone();
 
     // Check generated data is up to date
-    for infra in infras.iter() {
-        generate::refresh(&conn, &infra, false)?;
-    }
+    let computation = thread::spawn(move || -> Result<(), Box<dyn Error + Send + Sync>> {
+        for infra in infras_clone.iter() {
+            generate::refresh(&conn, &infra, false)?;
+        }
+        Ok(())
+    });
```

Comment:
why run it in the background? it sounds like a pretty unsafe thing to do, as the database state is inconsistent in the meanwhile

Reply:
It might take several seconds to refresh all infra. Meanwhile the service is considered as unhealthy. I fear that k8s kill the service.

Reply:
alright then, but this needs locking